akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonProxy.scala

```diff
@@ -203,8 +201,9 @@ final class ClusterSingletonProxy(singletonManagerPath: String, settings: Cluste
    */
   def add(m: Member): Unit = {
     if (matchingRole(m))
-      trackChange {
-        () ⇒ membersByAge += m
+      trackChange { () ⇒
+        membersByAge -= m // replace
```

Comment:
For consistency?

~~~scala
membersByAge = membersByAge - m + m // replace
~~~